Automattic/jetpack
# JS Components: Convert JetpackFooter component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `JetpackFooter` component to TypeScript.
* It also converts the `JetpackFooter` stories and unit tests to TS
* Adds more test cases for `JetpackFooter` unit tests

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `JetpackFooter` component work as expected
* Check the consumers of `JetpackFooter` and related components and confirm that they work as expected
* Check for any regression


<img width="1173" alt="Screenshot 2022-08-01 at 1 37 26 PM" src="https://user-images.githubusercontent.com/18226415/182102879-c12ed04e-8bb2-4a7f-a549-ebea2a561bc1.png">



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314264001
